### DISCUSSION TITLE
About uninstall registration

### DISCUSSION THREAD

**Discussion by ad-chuliang** [wixtoolset/issues#8653] [Questions]
I have a maintainess dialog when I open .msi after installation. If I double click .msi directly, the maintain dialog will show correctly. But, when I want to uninstall exe in the Control Panel\Programs\Programs and Features, it just uninstall without any operation in need. So, I want to have the same effect of openning .msi directly. In my investigation, if the installation package is 64-bit, its uninstall information is in the registry under the path HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\ . I tried to write the data using the RegistryValue element, and the data corresponding to the name ‘UninstallString’ from ‘MsiExec.exe /X{11D32F7D-DF4D-46C3-81CF- 65D219A4ED9C}’ to “MsiExec.exe /i{11D32F7D-DF4D-46C3-81CF-65D219A4ED9C}”. Both are invalid.

**Comment by bevanweiss** [comment]
What does your authoring look like?
Are you setting the [ARPNOMODIFY](https://learn.microsoft.com/en-us/windows/win32/msi/arpnomodify) property?

If you're using certain UI templates then it might get set by them, in which case you'd want to consider your UI selection, and override it back to your desired state as appropriate
https://github.com/wixtoolset/UI.wixext/blob/0e7fa93306858a9adde5b64ede920492632ba9e6/src/wixlib/WixUI_Minimal.wxs#L58

**Reply by ad-chuliang** [reply]
I copied WixUI_InstallDir.wxs. after that I changed the UI Id and made some modifications for some custom actions. Of course, "<Property Id="ARPNOMODIFY" Value="1" />" in it. I don't know about the role of ARPNOMODIFY actually. Finally, I found the path of Registry，where wix auto write without my any code, and the path is： 
Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Installer\UserData\S-1-5-18\Products\D7F23D11D4FD3C6418FC562D914ADEC9\InstallProperties.

**Reply by bevanweiss** [reply]
If you're writing a Windows Installer, you probably need to know something about the Windows Installer system.
So you WILL need to 'know about the role of ARPNOMODIFY actually'... ;)

**Reply by ad-chuliang** [reply]
Now, I get the use of ARPNOMODIFY. However, it just work on right click menu. When I do not set ARPNOMODIFY = 1 and right click software in the control panel, there is modify operation on the menu, but still no maintain dialog show up.
